app/src/main/java/org/thoughtcrime/securesms/onboarding/LandingActivity.kt

```diff
@@ -19,28 +81,168 @@ class LandingActivity : BaseActionBarActivity() {
+            .let(::setContentView)
+
         IdentityKeyUtil.generateIdentityKeyPair(this)
         TextSecurePreferences.setPasswordDisabled(this, true)
         // AC: This is a temporary workaround to trick the old code that the screen is unlocked.
         KeyCachingService.setMasterSecret(applicationContext, Object())
     }
 
-    private fun register() {
-        val intent = Intent(this, RegisterActivity::class.java)
-        push(intent)
+    @Preview
+    @Composable
+    private fun LandingScreen(
+        @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+    ) {
+        PreviewTheme(themeResId) {
+            LandingScreen()
+        }
+    }
+
+    @OptIn(ExperimentalFoundationApi::class)
+    @Composable
+    private fun LandingScreen() {
+
+        var count by remember { mutableStateOf(0) }
+        val listState = rememberLazyListState()
+
+        LaunchedEffect(Unit) {
+            delay(500.milliseconds)
+            while(count < MESSAGES.size) {
+                count += 1
+                listState.animateScrollToItem(0.coerceAtLeast((count - 1)))
+                delay(1500L)
+            }
+        }
+
+        Column(modifier = Modifier.padding(horizontal = 36.dp)) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                stringResource(R.string.onboardingBubblePrivacyInYourPocket),
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                style = MaterialTheme.typography.h4,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(24.dp))
+
+            LazyColumn(
+                state = listState,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+                items(
+                    MESSAGES.take(count),
+                    key = { it.stringId }
+                ) { item ->
+                    MessageText(
+                        stringResource(item.stringId),
+                        item.isOutgoing,
+                        modifier = Modifier.animateItemPlacement()
+                    )
+                }
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            OutlineButton(
+                text = stringResource(R.string.onboardingAccountCreate),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_create_account_button)
+            ) { startPickDisplayNameActivity() }
+            Spacer(modifier = Modifier.height(14.dp))
+            FilledButton(
+                text = stringResource(R.string.onboardingAccountExists),
+                modifier = Modifier
+                    .width(262.dp)
```

Comment:
Are we hardcoding width?
What happens on  different form factors / screen sizes?

Reply:
The designs don't really specify. I was thinking it was perhaps a bit premature to extract it somewhere before I know more.

e.g. I don't knnow if the margins are the important part, or the widths, nor when we would reach some max width... perhaps it should be `widthIn(max = wideButtonWidth).margin(horizontal(mediumMargin)` and then probably buried in some `Button()` style or something, but not 100% on the design language, so wrote it like this for now.

Reply:
I think it'd be great to start coming up with standards and rules for this kind of things. Do we have a designer to check those decisions against?
Buttons should be pretty standardised in the app, including their behaviour. We should make sure we have standardised, reusable components, and making sure they properly cater for the various form factors and text size etc..